File: backend/server/model/post.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type PostStatus string

const (
	PostStatusPending  PostStatus = "pending"
	PostStatusApproved PostStatus = "approved"
	PostStatusRejected PostStatus = "rejected"
	PostStatusSold     PostStatus = "sold"
	PostStatusHidden   PostStatus = "hidden"
	PostStatusDeleted  PostStatus = "deleted"
)

type Category struct {
	ID               string  `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	ParentCategoryID *string `gorm:"type:uuid;index;constraint:OnDelete:CASCADE;" json:"parent_category_id"`
	Name             string  `json:"name"`

	ParentCategory *Category `json:"parent_category,omitempty"`
}

type Post struct {
	ID          string         `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	UserID      *string        `gorm:"type:uuid" json:"user_id"`
	CategoryID  *string        `gorm:"type:uuid" json:"category_id"`
	AddressID   *string        `gorm:"type:uuid" json:"address_id"`
	Title       string         `json:"title"`
	Description string         `json:"description"`
	Price       uint           `json:"price"`
	Status      PostStatus     `json:"status"`
	SoldAt      *time.Time     `json:"sold_at,omitempty"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"deleted_at,omitempty"`

	User     *User     `json:"user,omitempty"`
	Category *Category `json:"category,omitempty"`
	Address  *Address  `json:"address,omitempty"`
}

type PostImage struct {
	PostID     *string `gorm:"type:uuid;primaryKey" json:"post_id"`
	ImageURL   string  `gorm:"primaryKey" json:"image_url"`
	ImageOrder uint    `json:"image_order"`
}

File: frontend/ReSell/app/src/main/java/com/example/resell/model/Post.kt
package com.example.resell.model

import com.squareup.moshi.Json
import java.time.LocalDateTime
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Post(
    @Json(name = "id") val id: String,
    @Json(name = "user_id") val userID: String,
    @Json(name = "category_id") val categoryID: String,
    @Json(name = "ward_id") val wardID: String,
    @Json(name = "title") val title: String,
    @Json(name = "description") val description: String? = null,
    @Json(name = "price") val price: Int,
    @Json(name = "status") val status: PostStatus,
    @Json(name = "created_at") val createdAt: LocalDateTime? = null,
    @Json(name = "updated_at") val updatedAt: LocalDateTime? = null,
    @Json(name = "deleted_at") val deletedAt: LocalDateTime? = null,
    @Json(name = "sold_at") val soldAt: LocalDateTime? = null,
    @Json(name = "user") val user: User? = null,
    @Json(name = "category") val category: Category? = null,
    @Json(name = "ward") val ward: Ward? = null,
    @Json(name = "post_images") val images: List<PostImage>? = null
)

@JsonClass(generateAdapter = true)
data class PostData(
    @Json(name = "id") val id: String,
    @Json(name = "title") val title: String,
    @Json(name = "status") val status: String,
    @Json(name = "category") val category: String,
    @Json(name = "owner") val owner: String,
    @Json(name = "fullname") val fullname: String,
    @Json(name="user_id") val userId:String,
    @Json(name = "avatar") val avatar: String,
    @Json(name = "description") val description: String,
    @Json(name = "thumbnail") val thumbnail: String,
    @Json(name = "address") val address: String,
    @Json(name = "price") val price: Int,
    @Json(name = "created_at") val createdAt: LocalDateTime?=null,
    @Json(name = "is_following") val isFollowing: Boolean,
    @Json(name = "is_favorite") val isFavorite: Boolean,
    @Json(name = "images") var images: List<PostImage>? = emptyList()
)


@JsonClass(generateAdapter = false)
enum class PostStatus {
    @Json(name = "pending")
    PENDING,
    @Json(name = "approved")
    APPROVED,
    @Json(name = "rejected")
    REJECTED,
    @Json(name = "sold")
    SOLD
}